### PR TITLE
refactor: use near_network_primitives directly

### PR DESCRIPTION
**Pull request by pmnoxx**
Hey, this is going to be an epic change, but it will improve code quality.

We currently have a package called `near-network-utils`. For some reason we are important all types through `near-network`.
From not on we will import all packages directly.

Also `blacklist_from_iter` got moved out of `near-network`. It's not even used there.

The final goal is to remove dependency on `near-network` from packages that don't need it.

### CODE REVIEW

File: nearcore/src/config.rs [hunk 1]
```diff
@@ -36,6 +37,45 @@ use near_primitives::version::PROTOCOL_VERSION;
...
 use near_telemetry::TelemetryConfig;
 
+/// `blacklist_from_iter` - converts list of addresses represented by strings to <IpAddr, BlockedPorts> HashMap
+///
+/// Arguments:
+/// - `blacklist`- list of strings in following formats:
+///    - "IP" - for example 127.0.0.1 - if only IP is provided we will block all ports
+///    - "IP:PORT - for example 127.0.0.1:2134
+pub fn blacklist_from_iter<T>(blacklist: T) -> HashMap<IpAddr, BlockedPorts>
+where
```
Comment (bowenwang1996) [inline nearcore/src/config.rs:47]: I am not sure whether it makes sense for this function to be in this file. It feels like a function specific to network

Comment (pmnoxx) [reply nearcore/src/config.rs:47]: @bowenwang1996 It doesn't. I removed it out and moved it out to another package in another PR.

Comment (pmnoxx) [reply nearcore/src/config.rs:47]: @bowenwang1996 This function is not used in network at all. It's used only for parsing config. Though we can put it inside near-network-primitives if you think that's better.

Comment (bowenwang1996) [reply nearcore/src/config.rs:47]: yeah that might be better

File: chain/network-primitives/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+use crate::types::{BlockedPorts, PatternAddr};
+use std::collections::{HashMap, HashSet};
+use std::net::IpAddr;
+
+/// `blacklist_from_iter` - converts list of addresses represented by strings to <IpAddr, BlockedPorts> HashMap
```
Comment (mm-near) [inline chain/network-primitives/src/utils.rs:5]: nit: you don't need to repeat the name of the function ('blacklist_from_iter')

**Review by mm-near — APPROVED** [review]

**Comment by pmnoxx** [comment]
@frol @chefsale @bowenwang1996 Can you take a look?

**Review by chefsale — APPROVED** [review]

**Comment by frol** [comment]
I delegate my vote to @miraclx since he has more context around -primitive crates

**Review by miraclx — APPROVED** [review]
LGTM

**Pull request merged**